tests/migration_test.go

```diff
@@ -4552,8 +4553,72 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 			}
 		})
 	})
+	Context("ResourceQuota rejection", func() {
+		It("Should contain condition when migrating with quota that doesn't have resources for both source and target", func() {
+			vmiRequest := resource.MustParse("200Mi")
+			vmi := libvmi.NewCirros(
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithResourceMemory(vmiRequest.String()),
+			)
+
+			vmiRequest.Add(resource.MustParse("50Mi")) //add 50Mi memoryOverHead to make sure vmi creation won't be blocked
+			enoughMemoryToStartVmiButNotEnoughForMigration := services.GetMemoryOverhead(vmi, runtime.GOARCH, nil)
+			enoughMemoryToStartVmiButNotEnoughForMigration.Add(vmiRequest)
+			resourcesToLimit := k8sv1.ResourceList{
+				k8sv1.ResourceMemory: resource.MustParse(enoughMemoryToStartVmiButNotEnoughForMigration.String()),
+			}
+
+			By("Creating ResourceQuota with enough memory for the vmi but not enough for migration")
+			resourceQuota := newResourceQuota(resourcesToLimit, testsuite.GetTestNamespace(vmi))
+			_ = createResourceQuota(resourceQuota, 90*time.Second)
+
+			By("Starting the VirtualMachineInstance")
+			_ = tests.RunVMIAndExpectLaunch(vmi, 240)
+
+			By("Trying to migrate the VirtualMachineInstance")
+			migration := tests.NewRandomMigration(vmi.Name, testsuite.GetTestNamespace(vmi))
+			migration = tests.RunMigration(virtClient, migration)
+			Eventually(func() error {
+				migration, migerr := virtClient.VirtualMachineInstanceMigration(migration.Namespace).Get(migration.Name, &metav1.GetOptions{})
+				if migerr != nil {
+					return migerr
+				}
+				for _, cond := range migration.Status.Conditions {
+					if cond.Type == v1.VirtualMachineInstanceMigrationRejectedByResourceQuota {
+						return nil
+					}
+				}
+				return fmt.Errorf("migration doesn't contain migrationRejectedByResourceQuota condition")
+			}, 60*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+		})
+	})
 })
 
+func createResourceQuota(resourceQuota *k8sv1.ResourceQuota, timeout time.Duration) *k8sv1.ResourceQuota {
+	virtCli := kubevirt.Client()
+
+	var obj *k8sv1.ResourceQuota
+	var err error
+	EventuallyWithOffset(1, func() error {
```

Comment:
Maybe the Api server won't be responsive 

> > > I have left a few suggestions on the tests.
> > > @vladikr I think the 3rd commit is unnecessary. We could just re-queue when we see quota rejection
> > 
> > 
> > but in case nothing changes in the cluster we would just keep re-queuing and that is Inefficient , the only thing that could unblock that migration is resourceQuota deletion or resourceQuota.status.hard\ resourceQuota.spec.hard update .
> 
> As far as understand it will still happen, you did not address it. Still, we should have a timeout that will be triggered.

it will happen but with back-off logic that could delay the migration in case of status change

Reply:
I removed the timeout.
> Please don't ignore any errors. We should not program defensively here.
can you please elaborate?

Reply:
The `Create` call should succeed. Please remove the `Eventually` block.